cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -46,8 +72,14 @@ private[lagom] object JoinClusterImpl {
           // since that would have blocked the shutdown of the ActorSystem.
           val t = new Thread(new Runnable {
             override def run(): Unit = {
-              if (Try(Await.ready(system.whenTerminated, 10.seconds)).isFailure)
-                System.exit(-1)
+              if (Try(Await.ready(system.whenTerminated, 10.seconds)).isFailure) {
+                println("Halting JVM.")
+                Runtime.getRuntime.halt(EXIT_TIMEOUT_WITH_HALT)
+              }
+              if (needsExiting) {
+                println("Proceed to JVM shutdown with exit status: " + exitStatus.get())
```

Comment:
The next line has `EXIT_TIMEOUT` hardcoded as the exit status, to this message is inaccurate.